appinfo/routes.php

```diff
@@ -94,8 +94,11 @@
 ['name' => 'item_api#read_multiple', 'url' => '/api/v1-2/items/read/multiple', 'verb' => 'PUT'], // FIXME: POST would be more correct
 ['name' => 'item_api#unread_multiple', 'url' => '/api/v1-2/items/unread/multiple', 'verb' => 'PUT'], // FIXME: POST would be more correct
 ['name' => 'item_api#star', 'url' => '/api/v1-2/items/{feedId}/{guidHash}/star', 'verb' => 'PUT'], // FIXME: POST would be more correct
+['name' => 'item_api#star_by_id', 'url' => '/api/v1-2/items/{itemId}/star', 'verb' => 'PUT'], // FIXME: POST would be more correct
 ['name' => 'item_api#unstar', 'url' => '/api/v1-2/items/{feedId}/{guidHash}/unstar', 'verb' => 'PUT'], // FIXME: POST would be more correct
+['name' => 'item_api#unstar_by_id', 'url' => '/api/v1-2/items/{itemId}/unstar', 'verb' => 'PUT'], // FIXME: POST would be more correct
 ['name' => 'item_api#star_multiple', 'url' => '/api/v1-2/items/star/multiple', 'verb' => 'PUT'], // FIXME: POST would be more correct
+['name' => 'item_api#star_multiple_by_id', 'url' => '/api/v1-2/items/star/multiple_by_id', 'verb' => 'PUT'], // FIXME: POST would be more correct
 ['name' => 'item_api#unstar_multiple', 'url' => '/api/v1-2/items/unstar/multiple', 'verb' => 'PUT'], // FIXME: POST would be more correct
```

Comment:
I believe that this would be more RESTful

```suggestion
['name' => 'item_api#star_multiple_by_id', 'url' => '/api/v1-2/items/star', 'verb' => 'PUT'], // FIXME: POST would be more correct
['name' => 'item_api#unstar_multiple', 'url' => '/api/v1-2/items/unstar', 'verb' => 'PUT'], // FIXME: POST would be more correct
```

Reply:
As far as I understand, changing `PUT` + `/api/v1-2/items/unstar/multiple` to `PUT` + `/api/v1-2/items/unstar` would break the backward compatibility. No doubt it's more RESTful but I'm not sure it worth breaking the API.

Please note that `/api/v1-2/items/star/multiple` and `/api/v1-2/items/unstar/multiple` are existing endpoints. My suggested endpoints just mimic them and add a postfix `_by_id` to avoid any conflicts. I have nothing against mapping those new endpoints to the root path, it you think it won't cause any issues with existing mappings. That raises the question why the existing `/multiple` endpoints don't use this path though.

Reply:
I didn't see they existed yet, please do map the new method to the root indeed, I didn't write the original API implementation so I have no idea why it is the way it is 😄

Reply:
@SMillerDev  FYI: I guess it's because the corresponding `/read` and `/unread` endpoints have a different meaning. When you call `PUT` on them, it means read/unread **all** of the items (they don't expect the array of IDs as their `/multiple` counterparts do). Probably the Nextcloud mapper isn't/wasn't smart enough to handle both of those actions on the same endpoint which led to this `/multiple` hack for the function that accepts the array of IDs and don't simply changes all of the items in the DB as the root endpoint does. Adding different behavior to otherwise similar `/star` and `/unstar` endpoints is far from ideal but I don't mind it since the other option is not ideal too.